reviewrot/irc.py

```diff
@@ -0,0 +1,68 @@
+import socket
+import logging
+import time
+
+log = logging.getLogger(__name__)
+
+
+class IRC:
+    """
+    IRC component used to connect to channel and send
+    message to channel
+    """
+
+    def __init__(self, config):
+        self.irc = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.server = config['server']
+        self.port = config['port']
+        self.channel = config['channel']
+        self.nick = 'review_rot_bot'
+
+    def connect(self):
+        """
+        Connects the bot to the IRC channel
+        """
+        log.debug('Connecting to server')
+        self.irc.connect((self.server, self.port))
+
+        log.debug('Joining channel')
+        self.irc.send(
+            'USER {} {} {} {}\n'.format(
+                self.nick, self.nick, self.nick, self.nick
+            ).encode('utf-8')
+        )
+        self.irc.send('NICK {}\n'.format(self.nick).encode('utf-8'))
+        self.irc.send('JOIN {}\n'.format(self.channel).encode('utf-8'))
+
+        # IRC servers use ping for authorization
+        buffer = self.irc.recv(1024)
+        msg = buffer.split()
+        if msg[0] == "PING":
+            self.irc.send('PONG {}\n'.format(msg[1]).encode('utf-8'))
+
+        log.debug('Connected')
+
+    def send_msg(self, msg):
+        """
+        Sends message to IRC channel
+        msg (string): message to be sent to channel
+        """
+        log.debug('sending msg' + msg)
+        return self.irc.send(
+            'PRIVMSG {} :{}\n'.format(self.channel, msg).encode('utf-8')
+        )
+        # to avoid flooding the channel
+        time.sleep(1)
+
+    def quit(self):
+        """
+        Disconnect bot from channel and close the connection
+        """
+        log.debug('closing connection')
+        self.irc.send('QUIT \n'.encode('utf-8'))
+        buffer = self.irc.recv(1024)
+        # wait for server to recieve all messages
+        # and acknowledge the client quit
```

Comment:
What if doesn't acknowledge?

Reply:
it's mostly for the messages and so that the irc client prints 
`review_rot_bot left (Client Quit)`
instead of 
`review_rot_bot left (Read error: Connection reset by peer)`